addons/account/models/account_move.py

```diff
@@ -2734,6 +2727,12 @@ def _post(self, soft=True):
                     to_write['line_ids'].append((1, line.id, {'name': to_write['payment_reference']}))
                 move.write(to_write)
 
+        for move in to_post\
+                .filtered(lambda mv: mv.restrict_mode_hash_table and not (mv.secure_sequence_number or mv.inalterable_hash))\
+                .sorted(lambda mv: (mv.date, mv.ref or '', mv.id)):
+            new_number = move.journal_id.secure_sequence_id.next_by_id()
+            move.write({'secure_sequence_number': new_number, 'inalterable_hash': move._get_new_hash(new_number)})
```

Comment:
Why do you move it? What happens if you write `state: posted` via RPC instead of calling `action_post`?
This could be a computed field depending on the state and `INTEGRITY_HASH_{MOVE,LINE}_FIELDS`
It would also help you check that it doesn't change based on the dependencies

When you compute, if:
* it is posted and you don't have a hash: assign the hash value
* it is posted and you have a hash: check that it is the same and do nothing
* it is not posted and you have a hash: raise error
* it is not posted and you don't have a hash: do nothing

Reply:
Then you don't need any code in `write`, `_post`, `button_draft` or `button_cancel`